developer/services/qrexec.md

```diff
@@ -0,0 +1,307 @@
+Thus, the framework allows dom0 to be the RPC target as well.
+
+Thanks to the framework, RPC programs are very simple -- both RPC client and server just use their stdin/stdout to pass data.
+The framework does all the inner work to connect these processes to each other via `qrexec-daemon` and `qrexec-agent`.
+Additionally, disposable VMs are tightly integrated -- RPC to a DisposableVM is identical to RPC to a normal domain, all one needs is to pass `$dispvm` as the remote domain name.
+
+## Qubes RPC administration
+
+<!-- (*TODO: fix for non-linux dom0*) -->
+
+The dom0 directory `/etc/qubes-rpc/policy/` contains files for each available RPC action.
+Together their contents make up the RPC access policy database.
+Currently, the defined actions are:
+
+    qubes.ClipboardPaste
+    qubes.Filecopy
+    qubes.GetImageRGBA
+    qubes.GetRandomizedTime
+    qubes.Gpg
+    qubes.GpgImportKey
+    qubes.InputKeyboard
+    qubes.InputMouse
+    qubes.NotifyTools
+    qubes.NotifyUpdates
+    qubes.OpenInVM
+    qubes.OpenURL
+    qubes.PdfConvert
+    qubes.ReceiveUpdates
+    qubes.SyncAppMenus
+    qubes.USB
+    qubes.VMShell
+    qubes.WindowIconUpdater
+
+These files contain lines with the following format:
+
+    srcvm destvm (allow|deny|ask)[,user=user_to_run_as][,target=VM_to_redirect_to]
+
+You can specify srcvm and destvm by name or by one of three reserved keywords: `@anyvm`, `@dispvm`, and `dom0` (without the `@`).
+Only `@anyvm` keyword makes sense in srcvm field.
+(Service calls from dom0 are currently always allowed, `@dispvm` means "new VM created for this particular request," so it is never a source of request.)
+Whenever a RPC request for an action is received, the domain checks the first matching line of the relevant file in `/etc/qubes-rpc/policy/` to determine access:
+whether to allow the request, what VM to redirect the execution to, and what user account the program should run under.
+Note that if the request is redirected (`target=` parameter), policy action remains the same - even if there is another rule which would otherwise deny such request.
+If the policy file does not exist, the user is prompted to create one.
+If still there is no policy file after prompting, the action is denied.
+
+In the target VM, the `/etc/qubes-rpc/RPC_ACTION_NAME` must exist, containing the file name of the program that will be invoked, or being that program itself - in which case it must have executable permission set (`chmod +x`).
+
+In the src VM, one should invoke the client via:
+
+    $ qrexec-client-vm target_vm_name RPC_ACTION_NAME rpc_client_path client arguments
+
+Note that only stdin/stdout is passed between RPC server and client -- notably, no command line arguments are passed.
+Source VM name is specified by `QREXEC_REMOTE_DOMAIN` environment variable.
+By default, stderr of client and server is logged to respective `/var/log/qubes/qrexec.XID` files.
```

Comment:
This is no longer true - it's logged to a local syslog/journald in the VM where process is running.